whylog/config/rule.py

```diff
@@ -19,10 +19,11 @@ def serialize(self):
             "constraints": self._constraints,
         }
 
-    def get_new_parsers(self, old_parsers):
+    def get_new_parsers(self, parser_name_generator):
         new_parsers = []
         for parser in itertools.chain([self._effect], self._causes):
-            if parser.name not in old_parsers:
+            #TODO: Refactor if teachers are mulithreding
+            if parser_name_generator.is_free_parser_name(parser.name, set()):
```

Comment:
You probably don't need to create a new empty set every time this is called. Maybe use an empty frozenset in class scope?